src/create/aws/create-queue/index.js

```diff
@@ -4,26 +4,67 @@ var aws = require('aws-sdk')
 
 /**
  * creates two queues
+ * optional: creates associated deadletter queue when maxReceiveCount is specified
  *
  * - appname-staging-queue
  * - appname-production-queue
+ *
  */
+
 module.exports = function _createSQS(params, callback) {
 
   assert(params, {
     app: String,
-    queue: String
+    queue: typeof params.queue === 'object' ? Object : String,
   })
 
+  var name = params.queue
+  var attrs = {}
+  var sqs = new aws.SQS({ region: process.env.AWS_REGION })
+
+  if (typeof params.queue === 'object') {
+    name = Object.keys(params.queue)[0] // myqueuename
+    attrs = params.queue[name] // { maxReceiveCount: 10 }
+  }
+
+  function createDeadLetterQueue(env, callback, queueName) {
+    sqs.createQueue({ QueueName: `${queueName}-deadletter` },
+      function _createQueue(err, data) {
+        if (err) {
+          console.log(err)
+          callback();
+        } else {
+          var params = {
+            QueueUrl: data.queueUrl,
+            AttributeNames: ['QueueArn']
+          }
+          sqs.getQueueAttributes(params, function _getQueueAttributes(err, data) {
+            if (err) {
+              console.log(err);
+              callback();
+            }
+            callback(data.Attributes.QueueArn);
```

Comment:
If you pass a value as the first param to a callback thats usually considered an error in Node…also the line above would execute the callback (if err is defined) and then ALSO execute this callback.